lib/AST/NameLookup.cpp

```diff
@@ -800,6 +800,77 @@ static CanType removeThrownError(Type type) {
   })->getCanonicalType();
 }
 
+/// When `SendableCompletionHandlers` feature is enabled, ClangImporter
+/// is going to inject `@Sendable` attribute into the completion handler
+/// parameter's function type. We need to match this behavior in Swift
+/// declarations that could shadow ObjC declarations otherwise it would
+/// result in ambiguities.
+static CanType injectSendableToCompletionHandler(ValueDecl *decl,
+                                                 CanType type) {
+  auto &ctx = decl->getASTContext();
+
+  if (!ctx.LangOpts.hasFeature(Feature::SendableCompletionHandlers))
+    return type;
+
+  if (!decl->isObjC() || decl->isObjCImplementation())
+    return type;
+
+  auto *paramList = getParameterList(decl);
+  if (!paramList || paramList->size() == 0)
+    return type;
+
+  // Check if the last parameter expects "completionHandler" argument.
+  auto *lastParam = paramList->back();
+  if (!isCompletionHandlerParamName(lastParam->getArgumentName().str()))
+    return type;
+
+  return type
+      .transformRec([](TypeBase *type) -> std::optional<Type> {
+        auto funcType = type->getAs<AnyFunctionType>();
+        if (!funcType || funcType->isAsync() || funcType->getNumParams() == 0)
+          return std::nullopt;
+
+        auto &lastParam = funcType->getParams().back();
+        if (!(lastParam.hasLabel() &&
+              isCompletionHandlerParamName(lastParam.getLabel().str())))
+          return std::nullopt;
+
+        auto completionType = lastParam.getPlainType();
+        // Completion handler could be wrapped in a number of optionals.
+        auto newCompletionType = completionType.transformRec(
+            [](TypeBase *type) -> std::optional<Type> {
+              if (auto *funcType = type->getAs<AnyFunctionType>()) {
+                auto extInfo = funcType->getExtInfo();
+                if (!extInfo.isSendable())
```

Comment:
Hm, this is actually tricky because ObjC side could have a `@nonSendable` attribute but we cannot know that here...